### PR TITLE
feat: scatter plot, add zoom controls

### PR DESCRIPTION
**Pull request by shayh**
## Description

added the option to enable zoom controls, mouse wheel and drag, of scatter plots

## Motivation and Context

enhance the capabilities of scatter plots, useful for large amount of data points

## How Has This Been Tested?

tested on the sandbox, also added a test for the toggle.

## Screenshots (if appropriate):
image

## Types of changes

-   [ ] Refactor (changes the way we code something without changing its functionality)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [X] Latest master code has been merged into this branch
-   [X] No commented out code (if required, place // TODO above with explanation)
-   [X] No linting issues
-   [X] Build is successful
-   [X] Code follows the [API Guidelines](http://britecharts.github.io/britecharts/topics-index.html#toc5__anchor)
-   [ ] Updated the documentation
-   [X] Added tests to cover changes
-   [X] All new and existing tests passed

### CODE REVIEW

File: src/charts/scatter-plot/scatter-plot.js [hunk 1]
```diff
@@ -439,7 +445,63 @@ export default function module() {
...
+     * @return {void}
+     * @private
+     */
+    function initZoom() {
+        if (!enableZoom) {
+            return;
+        }
+
+        const zoom = d3Zoom();
+        zoom.scaleExtent([0.5, 20]) // This control how much you can unzoom (x0.5) and zoom (x20)
```
Comment (Golodhros) [inline src/charts/scatter-plot/scatter-plot.js:462]: Let's make these two constants MIN_ZOOM and MAX_ZOOM

File: src/charts/scatter-plot/scatter-plot.js [hunk 2]
```diff
@@ -439,7 +445,63 @@ export default function module() {
...
+        // This add an invisible rect on top of the chart area. This rect can recover pointer events: necessary to understand when the user zoom
+        svg.append('rect')
+            .attr('class', 'zoom')
+            .attr('width', chartWidth)
+            .attr('height', chartHeight)
+            .style('fill', 'none')
+            .style('pointer-events', 'all')
+            .attr(
+                'transform',
+                'translate(' + margin.left + ',' + margin.top + ')'
```
Comment (Golodhros) [inline src/charts/scatter-plot/scatter-plot.js:478]: Let's use here a string interpolation, so `${margin.left}, ....

File: src/charts/scatter-plot/scatter-plot.js [hunk 3]
```diff
@@ -439,7 +445,63 @@ export default function module() {
...
+     * @return {void}
+     * @private
+     */
+    function updateChartAfterZoom(data, index, elements) {
+        //update scale
+        const transform = zoomTransform(elements[0]);
+        xScale = transform.rescaleX(xOriginalScale);
+        yScale = transform.rescaleY(yOriginalScale);
+        //update axes
+        svg.select('.x-axis-group .axis.x').call(axisBottom(xScale));
```
Comment (Golodhros) [inline src/charts/scatter-plot/scatter-plot.js:494]: could we use the xAxis variable here?

File: src/charts/scatter-plot/scatter-plot.js [hunk 4]
```diff
@@ -439,7 +445,63 @@ export default function module() {
...
+        svg.select('.y-axis-group .axis.y').call(axisLeft(yScale));
+        // update circle position
+        svg.select('.chart-group')
+            .selectAll('circle')
+            .attr('cx', (d) => xScale(d.x))
+            .attr('cy', (d) => yScale(d.y));
+        // update highlight location
```
Comment (Golodhros) [inline src/charts/scatter-plot/scatter-plot.js:501]: Nit: let's add a space between these blocks of code for increased readability

**Review by Golodhros — COMMENTED** [review]
Looks good in general, thanks for doing this!!

Just added some small comments.

**Comment by shayh** [comment]
@Golodhros sorry for the long delay, I fixed according to you comments, let me know if anything more is needed to push this enhancement forward.

**Pull request merged**